diff --git a/src/KOKKOS/pair_metatomic_kokkos.h b/src/KOKKOS/pair_metatomic_kokkos.h
--- a/src/KOKKOS/pair_metatomic_kokkos.h
+++ b/src/KOKKOS/pair_metatomic_kokkos.h
@@ -51,7 +51,5 @@ public:
     void store_forces(const at::Tensor& forces_tensor) override;
 
 private:
-    void pick_device(c10::Device& device, const char* requested) override;
-
     Kokkos::View<int32_t*, Kokkos::LayoutRight, DeviceType> type_mapping_kk;
 };
diff --git a/src/KOKKOS/pair_metatomic_kokkos.cpp b/src/KOKKOS/pair_metatomic_kokkos.cpp
--- a/src/KOKKOS/pair_metatomic_kokkos.cpp
+++ b/src/KOKKOS/pair_metatomic_kokkos.cpp
@@ -92,24 +92,5 @@ void PairMetatomicKokkos<DeviceType>::init_style() {
     }
 }
 
-template<class DeviceType>
-void PairMetatomicKokkos<DeviceType>::pick_device(torch::Device& device, const char* requested) {
-    device = KokkosDeviceToTorch<DeviceType>::convert();
-
-    if (requested != nullptr) {
-        auto requested_str = std::string(requested);
-        std::transform(requested_str.begin(), requested_str.end(), requested_str.begin(), ::tolower);
-        if (c10::DeviceTypeName(device.type(), /*lower_case=*/true) != requested_str) {
-            error->one(FLERR,
-                "requested device '{}' does not match the device being used by kokkos '{}', "
-                "use the non-kokkos version of this pair style to use a different "
-                "device for the model and LAMMPS",
-                requested, device.str()
-            );
-        }
-    }
-}
-
-
 template<class DeviceType>
 void PairMetatomicKokkos<DeviceType>::store_forces(const at::Tensor& forces_tensor) {
diff --git a/src/KOKKOS/fix_metatomic_kokkos.h b/src/KOKKOS/fix_metatomic_kokkos.h
--- a/src/KOKKOS/fix_metatomic_kokkos.h
+++ b/src/KOKKOS/fix_metatomic_kokkos.h
@@ -42,8 +42,6 @@ public:
    void final_integrate() override;
 
 private:
-   void pick_device(c10::Device& device, const char* requested) override;
-
    // Kokkos view for type mapping
    Kokkos::View<int32_t*, Kokkos::LayoutRight, DeviceType> type_mapping_kk;
 };
diff --git a/src/KOKKOS/fix_metatomic_kokkos.cpp b/src/KOKKOS/fix_metatomic_kokkos.cpp
--- a/src/KOKKOS/fix_metatomic_kokkos.cpp
+++ b/src/KOKKOS/fix_metatomic_kokkos.cpp
@@ -112,24 +112,5 @@ void FixMetatomicKokkos<DeviceType>::init() {
 /* ---------------------------------------------------------------------- */
 
-template<class DeviceType>
-void FixMetatomicKokkos<DeviceType>::pick_device(c10::Device& device, const char* requested) {
-    // Pick device based on Kokkos execution space
-    device = KokkosDeviceToTorch<DeviceType>::convert();
-
-    if (requested != nullptr) {
-        auto requested_str = std::string(requested);
-        std::transform(requested_str.begin(), requested_str.end(), requested_str.begin(), ::tolower);
-        if (c10::DeviceTypeName(device.type(), /*lower_case=*/true) != requested_str) {
-            error->all(FLERR,
-                "requested device '{}' does not match the device being used by kokkos '{}', "
-                "use the non-kokkos version of this fix to use a different "
-                "device for the model and LAMMPS",
-                requested, device.str()
-            );
-        }
-    }
-}
-
 /* ---------------------------------------------------------------------- */
 
 template<class DeviceType>
